web/l10n.js

```diff
@@ -0,0 +1,95 @@
+/* Copyright 2023 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/** @typedef {import("./interfaces").IL10n} IL10n */
+/**
+ * @implements {IL10n}
+ */
+class L10n {
+  #dir;
+
+  #l10n;
+
+  constructor({ lang, isRTL }, l10n = null) {
+    this.lang = L10n.#fixupLangCode(lang);
+    this.#l10n = l10n;
+    this.#dir = isRTL ?? L10n.#isRTL(this.lang) ? "rtl" : "ltr";
+  }
+
+  setL10n(l10n) {
+    this.#l10n = l10n;
+  }
+
+  /** @inheritdoc */
+  async getLanguage() {
+    return this.lang;
+  }
+
+  /** @inheritdoc */
+  async getDirection() {
+    return this.#dir;
+  }
+
+  /** @inheritdoc */
+  async get(ids, args = null, fallback) {
+    if (Array.isArray(ids)) {
+      return (await this.#l10n.formatMessages(ids.map(id => ({ id })))).map(
+        message => message.value
+      );
```

Comment:
This isn't a wonder of readability, given that sheer amount of parenthesis, so could we perhaps introduce a temporary variable to simplify this a little bit?